### PR TITLE
Boost: Add benefits interstitial

### LINKED ISSUE #23318: [Boost] Add benefits interstitial
Project: pc9hqz-iD-p2
Design: p1HpG7-f1L-p2 (please refer to Figma file for details)

Create the interstitial page with the pricing for, and benefits of Premium as per the Figma designs. Includes a link to the purchase / payments page which is already used in other Jetpack products.

### PR DESCRIPTION
**Pull request by haqadn**
Fixes #23318

#### Changes proposed in this Pull Request:
- [x] Useed svelte-navigator package for routing
- [x] Added benefits page
- [x] Added benefits page content and style
- [ ] Activated button to purchase plan

#### Jetpack product discussion
pc9hqz-iD-p2

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
- Go to 'Boost dashboard'
- Add `#/upgrade` to the URL
- Check the design on different screen sizes. dVyv3AO1fXG0gprXywMuMh-fi-2130%3A7342

image

### CODE REVIEW

File: projects/plugins/boost/app/assets/src/css/components/card.scss [hunk 1]
```diff
@@ -0,0 +1,52 @@
+.jb-card {
+  background: #FFFFFF url("../src/images/color-shift.svg") no-repeat top right;
```
Comment (thingalon) [inline projects/plugins/boost/app/assets/src/css/components/card.scss:2]: We don't include `/src/` in our distributed packages. Images from `/src/` typically get compiled into the JavaScript and embedded that way - but I don't believe CSS will do that. For images included via CSS, use the `assets/static/images` directory.

File: projects/plugins/boost/app/assets/src/css/components/checklist.scss [hunk 2]
```diff
@@ -0,0 +1,13 @@
+.jb-checklist {
+  li {
+	padding-top: $spacing-unit;
+	padding-bottom: $spacing-unit;
+	padding-left: 2em;
+	vertical-align: middle;
+	background: url("../src/images/check.svg") no-repeat;
```
Comment (thingalon) [inline projects/plugins/boost/app/assets/src/css/components/checklist.scss:7]: As above wrt static images.

File: projects/plugins/boost/app/assets/src/css/main/common.scss [hunk 3]
```diff
@@ -0,0 +1,14 @@
+@for $i from 1 through 3 {
```
Comment (thingalon) [inline projects/plugins/boost/app/assets/src/css/main/common.scss:1]: TIL about `@for` :)

File: projects/plugins/boost/app/assets/src/js/Main.svelte [hunk 4]
```diff
@@ -4,24 +4,31 @@
...
+
+	import routerHistory from './utils/router-history';
 </script>
 
-<div id="jb-admin-settings">
-	<div id="jb-settings" class="jb-settings">
-		<div class="jb-container">
-			<Header />
-		</div>
+<Router history={routerHistory}>
```
Comment (thingalon) [inline projects/plugins/boost/app/assets/src/js/Main.svelte:17]: This seems like a neat way of doing this 👍

File: projects/plugins/boost/app/assets/src/js/elements/BackButton.svelte [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<script>
```
Comment (thingalon) [inline projects/plugins/boost/app/assets/src/js/elements/BackButton.svelte:1]: 👍

File: projects/plugins/boost/app/assets/src/js/utils/create-hash-source.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { createHashHistory } from 'history';
+
+function createHashSource() {
```
Comment (thingalon) [inline projects/plugins/boost/app/assets/src/js/utils/create-hash-source.ts:6]: I don't understand what this is function is for - can we please have a docblock comment describing its purpose?

**Review by thingalon — CHANGES_REQUESTED** [review]
Wow, nice work - it's looking quite slick. :)

Be sure to check out the figma file in the i3 designs for box sizing: dVyv3AO1fXG0gprXywMuMh-fi-2130%3A7342

e.g.: The text "Optimize your website's performance" should be in a width-constrained box on it. If you click on elements in the figma, it shows you some CSS preview styles you can use as a starting location :)

I also notice there are a few warnings in the browser console:
image

There's also some text at the bottom of the page ("Special introductory pricing...") which is missing.

File: projects/plugins/boost/app/assets/src/css/components/card.scss (hunk 1 above)
Comment (eAdnan007) [reply projects/plugins/boost/app/assets/src/css/components/card.scss]: Updated

File: projects/plugins/boost/app/assets/src/css/components/checklist.scss (hunk 2 above)
Comment (eAdnan007) [reply projects/plugins/boost/app/assets/src/css/components/checklist.scss]: Updated

File: projects/plugins/boost/app/assets/src/js/utils/create-hash-source.ts (hunk 6 above)
Comment (eAdnan007) [reply projects/plugins/boost/app/assets/src/js/utils/create-hash-source.ts:12]: Added

**Comment by eAdnan007** [comment]
> I also notice there are a few warnings in the browser console: image

These warnings are there because the router is pushing the props. If we register the props, we will have warnings about unused props on every build/watch as a warning. If we ignore it, it will not appear for the production build.

> There's also some text at the bottom of the page ("Special introductory pricing...") which is missing.

Added

**Comment by eAdnan007** [comment]
> Be sure to check out the figma file in the i3 designs for box sizing: [dVyv3AO1fXG0gprXywMuMh-fi-2130%3A7342](https://href.li/?https://www.figma.com/file/dVyv3AO1fXG0gprXywMuMh/?node-id=2130%3A7342)
> 
> e.g.: The text "Optimize your website's performance" should be in a width-constrained box on it. If you click on elements in the figma, it shows you some CSS preview styles you can use as a starting location :)

I have kinda tangled this branch and #23319 by creating #23319 on top of this PR. The bounding box is correct in that branch as it is coming from a generic style which also modifes the structure of both cards. Can we merge this PR excluding this design fix and continue in the other PR?

**Comment by thingalon** [comment]
> These warnings are there because the router is pushing the props. If we register the props, we will have warnings about unused props on every build/watch as a warning. If we ignore it, it will not appear for the production build.

Fixed in 0d56820ef3.

**Review by thingalon — DISMISSED** [review]
Approved with the understanding that more CSS updates to come in a future PR.

**Review by thingalon — DISMISSED** [review]

**Review by thingalon — APPROVED** [review]

**Comment by eAdnan007** [comment]
r242769-wpcom

**Pull request merged**